runtime/interpreter/interpreter.go

```diff
@@ -840,7 +842,8 @@ func (interpreter *Interpreter) resultValue(returnValue Value, returnType sema.T
 		}
 	}
 
-	return NewEphemeralReferenceValue(interpreter, resultAuth(returnType), returnValue, returnType)
+	// result must be a resource to get here, so it will never be a reference
+	return NewEphemeralReferenceValue(interpreter, resultAuth(returnType), returnValue, returnType, EmptyLocationRange)
```

Comment:
Here and for similar calls: Maybe chop down the long argument list to multiple lines